concordium-consensus/src/Concordium/GlobalState/Persistent/Instances.hs

```diff
@@ -93,10 +94,11 @@ instance Applicative m => Cacheable m PersistentInstanceParameters
 data PersistentInstanceV (v :: Wasm.WasmVersion) = PersistentInstanceV {
     -- |The fixed parameters of the instance.
     pinstanceParameters :: !(BufferedRef PersistentInstanceParameters),
-    -- |The interface of 'pinstanceContractModule'. Note this is a BufferedRef to a Module as this
+    -- |The interface of 'pinstanceContractModule'. Note this is a 'HashedCachedRef' to a Module as this
     -- is how the data is stored in the Modules table. A 'Module' carries a BlobRef to the source
     -- but that reference should never be consulted in the scope of Instance operations.
-    pinstanceModuleInterface :: !(BufferedRef (Modules.ModuleV v)),
+    -- The module will always be of version @v@.
+    pinstanceModuleInterface :: !(HashedCachedRef Modules.ModuleCache Modules.Module),
```

Comment:
Wherever a `PersistentInstanceV` is constructed, I think it would be useful to have a brief argument that establishes why the module version will be `v`. (This would be the same as why the old `unsafeGetModuleReferenceV?` would be safe in the context.)

Reply:
@td202 I've added some more documentation. Can you please check that it is in line with what you would've expected?